odoo/addons/test_search_panel/tests/test_search_panel_select_range.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+import odoo.tests
+
+
+@odoo.tests.tagged('post_install', '-at_install', 'dam')
+class TestSelectRange(odoo.tests.SavepointCase):
+
+    def setUp(self):
+        super().setUp()
+        self.SourceModel = self.env['test_search_panel.source_model']
+        self.TargetModel = self.env['test_search_panel.category_target_model']
+        self.TargetModelNoParentName = self.env['test_search_panel.category_target_model_no_parent_name']
+
+    # Many2one
+
+    def test_many2one_empty(self):
+        result = self.SourceModel.search_panel_select_range('folder_id')
+        self.assertEqual(
+            result,
+            {
+                'parent_field': 'parent_id',
+                'values': [],
+            }
+        )
+
+    def test_many2one(self):
+        parent_folders = self.TargetModel.create([
+            {'name': 'Folder 1', },
+            {'name': 'Folder 2', },
+        ])
+
+        f1_id, f2_id = parent_folders.ids
+
+        children_folders = self.TargetModel.create([
+            {'name': 'Folder 3', 'parent_id': f1_id, },
+            {'name': 'Folder 4', 'parent_id': f2_id, },
+        ])
+
+        f3_id, f4_id = children_folders.ids
+
+        records = self.SourceModel.create([
+            {'name': 'Rec 1', 'folder_id': f1_id, },
+            {'name': 'Rec 2', 'folder_id': f3_id, },
+            {'name': 'Rec 3', 'folder_id': f4_id, },
+            {'name': 'Rec 4'},
+        ])
+
+        r1_id, _, r3_id, _ = records.ids
+
+        # counters
+        result = self.SourceModel.search_panel_select_range('folder_id')
+        self.assertEqual(
+            result['values'],
+            [
+                {'count': 2, 'display_name': 'Folder 1', 'id': f1_id, },
+                {'count': 1, 'display_name': 'Folder 2', 'id': f2_id, },
+                {'count': 1, 'display_name': 'Folder 3',
+                    'id': f3_id, 'parent_id': f1_id, },
+                {'count': 1, 'display_name': 'Folder 4',
+                    'id': f4_id, 'parent_id': f2_id, },
+            ]
+        )
+
+        # counters and search domain
+        kwargs = {'search_domain': [['id', 'in', [r1_id, r3_id]]], }
+        result = self.SourceModel.search_panel_select_range(
```

Comment:
result = self.SourceModel.search_panel_select_range(
            'folder_id', search_domain=[['id', 'in', [r1_id, r3_id]]])

to get rid of kwargs dict

Reply:
ah good! I thought it was weird. Cool

Reply:
done